modules/workers/nodepools.tf

```diff
@@ -29,6 +29,18 @@ resource "oci_containerengine_node_pool" "workers" {
       content {
         availability_domain = ad.value
         subnet_id           = each.value.subnet_id
+        dynamic "preemptible_node_config" {
+          for_each = each.value.preemptible.enable ? [1] : []
+          content {
+            dynamic "preemption_action" {
+              for_each = each.value.preemptible.enable ? [1] : []
```

Comment:
This seems redundant? With the `preemptible_node_config` block already dynamic on the value of `each.value.preemptible.enable`, I think this shouldn't need to be, e.g.:
```
preemption_action {
  type = "TERMINATE"
  is_preserve_boot_volume = each.value.preemptible.is_preserve_boot_volume
}
```